context/src/main/java/io/grpc/Deadline.java

```diff
@@ -164,13 +164,19 @@ public int compareTo(Deadline that) {
     return 0;
   }
 
-  /** Time source representing nanoseconds since fixed but arbitrary point in time. */
-  abstract static class Ticker {
+  /**
+   * Time source representing nanoseconds since fixed but arbitrary point in time.
+   *
+   * <p>This is public for testing purposes.
+   */
+  public interface DeadlineTicker {
     /** Returns the number of nanoseconds since this source's epoch. */
-    public abstract long read();
+    long read();
   }
 
-  private static class SystemTicker extends Ticker {
```

Comment:
I don't see any value avoiding creating the small singleton for testing. But meh.